cmd/demo/node_setup.go

```diff
@@ -109,63 +110,137 @@ func (n *node) setup() error {
 	return nil
 }
 
-// setupContracts reads from the config file whether the node should deploy or use
-// existing contract addresses.
-func (n *node) setupContracts() (err error) {
-	var adjAddr, assAddr common.Address
+func newTransactionContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+}
 
-	if config.Chain.Adjudicator == "deploy" {
-		adjAddr, err = deployAdjudicator(n.cb)
-		if err != nil {
-			return
-		}
-		fmt.Println("💭 Adjudicator contract deployed")
-	} else {
-		tmpAdj, err := strToAddress(config.Chain.Adjudicator)
-		if err != nil {
-			return err
-		}
-		adjAddr = ewallet.AsEthAddr(tmpAdj)
+func parseAndValidateAdjudicator(cb echannel.ContractBackend) (common.Address, error) {
+	var adjAddr common.Address
+	var err error
+
+	// parse address
+	adjAddr, err = stringToAddress(config.Chain.Adjudicator)
 
-		ctx, cancel := context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+	if err == nil {
+		// validate contract
+		ctx, cancel := newTransactionContext()
 		defer cancel()
-		if err := echannel.ValidateAdjudicator(ctx, n.cb, adjAddr); err != nil {
-			return errors.WithMessage(err, "validating adjudicator contract")
-		}
+		err = echannel.ValidateAdjudicator(ctx, cb, adjAddr)
+	}
 
-		fmt.Println("💭 Adjudicator contract validated")
+	return adjAddr, err
+}
+
+func parseAndValidateAssetHolder(cb echannel.ContractBackend, adjAddr common.Address) (common.Address, error) {
+	var assAddr common.Address
+	var err error
+
+	// parse address
+	assAddr, err = stringToAddress(config.Chain.Assetholder)
+
+	if err == nil {
+		// validate contract
+		ctx, cancel := newTransactionContext()
+		defer cancel()
+		err = echannel.ValidateAssetHolderETH(ctx, cb, assAddr, adjAddr)
 	}
 
-	if config.Chain.Assetholder == "deploy" {
-		assAddr, err = deployAsset(n.cb, adjAddr)
-		if err != nil {
+	return assAddr, err
+}
+
+func stringToAddress(s string) (common.Address, error) {
```

Comment:
If we do the parsing in the `config.go`, this function can also be moved there.

Reply:
done